Darwin/Services/StitchRemoteMongoDBService/StitchRemoteMongoDBServiceTests/SyncPerformanceIntTestHarness.swift

```diff
@@ -0,0 +1,341 @@
+                                                  update: ["$push": ["results": result.toBson()] as Document])
+                    }
+                }
+            }
+
+            if testParams.outputToStitch {
+                _ = outputColl?.updateOne(filter: ["_id": resultId],
+                                          update: ["$set": ["status": "Success"] as Document])
+            }
+        } catch {
+            let failureMessage = """
+            Failed on iteration \(iteration) of \(testParams.numIters)
+            with error \(String(describing: error))
+            """
+
+            print("PerfLog: Harness Error: \(failureMessage)")
+            if testParams.outputToStitch {
+                _ = outputColl?.updateOne(filter: ["_id": resultId],
+                                          update: ["$set": ["status": "Failed",
+                                                            "failureReason": failureMessage] as Document])
+            XCTFail(failureMessage)
+            }
+        }
+    }
+    // swiftlint:enable function_body_length
+}
+
+internal struct TestParams {
+    let testName: String
+    let runId: ObjectId
+    let numIters: Int
+    let numDocs: [Int]
+    let docSizes: [Int]
+    let dataProbeGranularityMs: Int
+    let numOutliersEachSide: Int
+    let stitchHostName: String
+    let outputToStdOut: Bool
+    let outputToStitch: Bool
+    let preserveRawOutput: Bool
+
+    init(
+        testName: String,
+        runId: ObjectId,
+        numIters: Int = 12,
+        numDocs: [Int] = [],
+        docSizes: [Int] = [],
+        dataProbeGranularityMs: Int = 1500,
+        numOutliersEachSide: Int = 1,
+        stitchHostName: String = "",
+        outputToStdOut: Bool = true,
+        outputToStitch: Bool = true,
+        preserveRawOutput: Bool = false
+    ) {
+        self.testName = testName
+        self.runId = runId
+        self.numIters = numIters
+        self.numDocs = numDocs
+        self.docSizes = docSizes
+        self.dataProbeGranularityMs = dataProbeGranularityMs
+        self.numOutliersEachSide = numOutliersEachSide
+        self.stitchHostName = stitchHostName
+        self.outputToStdOut = outputToStdOut
+        self.outputToStitch = outputToStitch
+        self.preserveRawOutput = preserveRawOutput
+    }
+
+    func toBson() -> Document {
+        return [
+            "testName": testName,
+            "runId": runId,
+            "numIters": numIters,
+            "dataProbeGranularityMs": dataProbeGranularityMs,
+            "numOutliersEachSide": numOutliersEachSide,
+            "stitchHostName": stitchHostName, // Todo: fix this
+            "date": Date(),
+            "sdk": "ios",
+            "results": []
+        ]
+    }
+}
+
+struct DoubleDataBlock {
+    var mean: Double = 0.0
+    var median: Double = 0.0
+    var max: Double = 0.0
+    var min: Double = 0.0
+    var stdDev: Double = 0.0
```

Comment:
[nit] all of these should be `let`s

Reply:
I cant do this in case you set numOutliers to be more than the number of iterations I just return these values